src/pipelines/vaccinations/ch_authority.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Any, Dict, List
+import requests
+from pandas import DataFrame
+from lib.data_source import DataSource
+from lib.utils import table_rename
+
+
+class SwitzerlandDataSource(DataSource):
+    def fetch(
+            self,
+            output_folder: Path,
+            cache: Dict[str, str],
+            fetch_opts: List[Dict[str, Any]],
+            skip_existing: bool = False,
+    ) -> Dict[str, str]:
+        # the url in the config is a json file which contains the actual url for vaccDosesAdministered
+        src_url = fetch_opts[0]['url']
+        data = requests.get(src_url).json()
+        vaccines_url = data['sources']['individual']['csv']['vaccDosesAdministered']
+        fetch_opts[0]['url'] = vaccines_url
+        return super().fetch(output_folder, cache, fetch_opts, skip_existing)
+
+    def parse_dataframes(
+        self, dataframes: Dict[Any, DataFrame], aux: Dict[str, DataFrame], **parse_opts
+    ) -> DataFrame:
+
+        # Rename the appropriate columns
+        data = table_rename(
+            dataframes[0],
+            {
+                "date": "date",
+                "geoRegion": "subregion1_code",
+                "sumTotal": "total_vaccine_doses_administered",
+            },
+            drop=True,
+        )
+
+        # Make sure all records have the country code and match subregion1 only
+        data["key"] = None
+        data["country_code"] = "CH"
+        data["subregion2_code"] = None
+        data["locality_code"] = None
+
+        # Country-level records have a known key
+        country_mask = data["subregion1_code"] == "CH"
```

Comment:
I think the file contains a subregion "CHFL" which will not be a match. If I'm not mistaken, it corresponds to the country of Liechtenstein (but please double check)

Reply:
both FL and CHFL don't match. FL I think should be Liechtenstein since 
a) the CHFL number is the sum of CH and FL 
b) the CHFL number corresponds to the Switzerland and Liechtenstein total at https://www.covid19.admin.ch/en/epidemiologic/vacc-doses

wdyt? Shall I mark the FL row as Liechtenstein?

Reply:
that is indeed the case! I see the below code in pipelines/epidemiology/ch_authority.py :

" # Principality of Liechtenstein is not in CH
        data.loc[data["subregion1_code"] == "FL", "key"] = "LI""

Reply:
added this row now

Reply:
Great!

Reply:
thanks!